File: Java/ascii.java
import java.util.Scanner;

public class ConversorASCII {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Digite um numero (0-127): ");
        int numero = scanner.nextInt();

        if (numero < 0 || numero > 127) {
            System.out.println("Valor invalido! Use 0 a 127.");
        } else {
            System.out.println("ASCII: " + (char) numero);
        }
        scanner.close();
    }
}

File: Fortran/ascii.f90
program numero_ascii
    implicit none
    integer :: numero
    character(len=1) :: letra

    write(*, '(A)', advance='no') 'Digite um numero (0-127): '
    read(*, *) numero

    if (numero < 0 .or. numero > 127) then
        write(*, *) 'Valor invalido! Use 0 a 127.'
    else
        letra = achar(numero)
        write(*, *) 'ASCII: ', letra
    end if

end program numero_ascii
